akka-http-core/src/main/scala/akka/http/javadsl/settings/WebSocketSettings.scala

```diff
@@ -0,0 +1,33 @@
+package akka.http.javadsl.settings
+
+import java.util.Random
+
+import akka.actor.ActorSystem
+import akka.annotation.DoNotInherit
+import akka.http.impl.settings.WebSocketSettingsImpl
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.Duration
+
+/**
+ * Public API but not intended for subclassing
+ */
+@DoNotInherit
+trait WebSocketSettings { self: WebSocketSettingsImpl ⇒
+  def randomFactory: () ⇒ Random
+  def periodicKeepAliveMode: String
+  def periodicKeepAliveMaxIdle: Duration
+
+  def withRandomFactory(newValue: () ⇒ Random): WebSocketSettings =
+    copy(randomFactory = newValue)
+  def withPeriodicKeepAliveMode(newValue: String): WebSocketSettings =
+    copy(periodicKeepAliveMode = newValue)
+  def withPeriodicKeepAliveMaxIdle(newValue: Duration): WebSocketSettings =
+    copy(periodicKeepAliveMaxIdle = newValue)
+}
+
+object WebSocketSettings extends SettingsCompanion[WebSocketSettings] {
+  override def create(config: Config): WebSocketSettings = WebSocketSettingsImpl(config)
+  override def create(configOverrides: String): WebSocketSettings = WebSocketSettingsImpl(configOverrides)
+  override def create(system: ActorSystem): WebSocketSettings = WebSocketSettingsImpl(system.settings.config)
+}
```

Comment:
I think sharing the settings between client and server looks ok. Is it likely we will need to add more one-side-ony settings in the future?

Reply:
I reminded myself the spec a bit more, and specifically more about the extension negotiation mechanisms https://tools.ietf.org/html/rfc7692 and I think it's also pretty symmetric... I looked at complexity of the added separate settings and they don't seem to be quite worth it. We'd figure out a way to solve it iff we'd need some asymetric part to it.